### PR TITLE
SIDM-6713 Add flux config for IDAM user dashboard

### PR DESCRIPTION
**Pull request by KianKwa**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SIDM-6713

### Change description ###

Add flux config for IDAM user dashboard

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: apps/idam/idam-user-dashboard/idam-user-dashboard.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      ingressHost: idam-user-dashboard.platform.hmcts.net
+      aadIdentityName: idam
+      useInterpodAntiAffinity: true
+      environment:
+        STRATEGIC_SERVICE_URL: https://idam-api.platform.hmcts.net
+        STRATEGIC_PUBLIC_URL: https://hmcts-access.service.gov.uk
+  chart:
+    spec:
+      chart: idam-user-dashboard
+      version: 0.2.8
```
Comment (nikola-naydenov-hmcts) [inline apps/idam/idam-user-dashboard/idam-user-dashboard.yaml:20]: Is this really your chart version?

Comment (KianKwa) [reply apps/idam/idam-user-dashboard/idam-user-dashboard.yaml]: this has been updated to 0.0.4

File: apps/idam/idam-user-dashboard/idam-user-dashboard.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        STRATEGIC_SERVICE_URL: https://idam-api.platform.hmcts.net
+        STRATEGIC_PUBLIC_URL: https://hmcts-access.service.gov.uk
+  chart:
+    spec:
+      chart: idam-user-dashboard
+      version: 0.0.4
+      sourceRef:
+        kind: HelmRepository
+        name: hmctspublic
+        namespace: flux-system
```
Comment (timja) [inline apps/idam/idam-user-dashboard/idam-user-dashboard.yaml:24]: is there a reason you aren't using the hmcts-charts repo?

File: apps/idam/idam-user-dashboard/preview.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: idam-user-dashboard
+  namespace: idam
+spec:
+  values:
+    nodejs:
+      image: hmctspublic.azurecr.io/idam/user-dashboard:prod-ae48887-20211206112137
```
Comment (timja) [inline apps/idam/idam-user-dashboard/preview.yaml:9]: is it intentional you aren't configuring any image policies to have this automatically deployed? applies throughout

Comment (KianKwa) [reply apps/idam/idam-user-dashboard/preview.yaml]: @timja Apology for the delay in replying. I was waiting to discuss with the current IDAM team on whether we need the special preview environment as this new application potentially need the to refer to the  new changes IDAM api on preview. We decided not to do that for now hence removing all the special policies

**Review by timja — APPROVED** [review]

**Review by nikola-naydenov-hmcts — APPROVED** [review]

**Pull request merged**